server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -234,6 +238,11 @@ public NodeStats(StreamInput in) throws IOException {
         } else {
             admissionControlStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            nodeCacheStats = in.readOptionalWriteable(NodeCacheStats::new);
+        } else {
+            nodeCacheStats = null;
+        }
```

Comment:
We should probably commit it in 2.x first before main instead of committing with 3.0 in main, backporting to 2.x and changing to 2.x in main

Reply:
Good point. Would it be easiest to just change this to V_2_14_0 in main and then backport to 2.x?

Reply:
I think if you commit this in 2.x first before main you'll break build on main since the bwc tests on main won't find the right field. I suppose if you get both PRs ready to be merged at the same time it may be fine but in the past what I've done is:
1. Merge into `main`
2. Merge into `2.x`
3. Merge another commit into `main` to update the version.

Reply:
> I think if you commit this in 2.x first before main you'll break build on main since the bwc tests on main won't find the right field.

Ah, interesting!

> Merge another commit into main to update the version.

I was thinking of how can be avoid this 2 step approach, but I guess there is no workaround unless you merge them together